src/Runtime/Distributions/Automata/Automaton.State.cs

```diff
@@ -152,16 +150,19 @@ public Transition[] GetTransitions()
             /// </param>
             /// <param name="group">The group of the added transitions.</param>
             /// <returns>The last state in the added transition series.</returns>
-            public State AddTransitionsForSequence(TSequence sequence, State destinationState = null, int group = 0)
+            public State AddTransitionsForSequence(TSequence sequence, State destinationState = default(State), int group = 0)
             {
-                State currentState = this;
-                IEnumerator<TElement> enumerator = sequence.GetEnumerator();
-                bool moveNext = enumerator.MoveNext();
-                while (moveNext)
+                var currentState = this;
```

Comment:
isn't that just a `foreach` loop?

Reply:
No, last element is handled in a different manner from others